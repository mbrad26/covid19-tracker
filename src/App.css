.App {
  text-align: center;
}

h2 {
  position: absolute;
  z-index: 10;
}
