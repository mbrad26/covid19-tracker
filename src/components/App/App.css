body {
  margin: 0px;
  padding: 0px;
  font-family: 'Space Mono', monospace;
  background-color: #00003f;
}

.container {
  min-width: 85.7vw;
}

#map-container {
  margin-top: 0.1em;
  padding: 5px;
}

#countries-table-container {
  margin-top: 0.1em;
  padding: 5px;
}

#data-cards-container {
  overflow: hidden;
  display: inline-flex;
}

@media only screen and (max-width: 767px) {

}
