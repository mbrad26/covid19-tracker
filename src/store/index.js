import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  applyMiddleware(saga, logger)
);

// console.log('STORE: ', store.getState());

saga.run(rootSaga);

export default store;
