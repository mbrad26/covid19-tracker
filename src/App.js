import React, { useState, useEffect } from 'react';
import ReactMapGl, { Marker } from 'react-map-gl';
import './App.css';

import axios from 'axios';

const url = 'https://covid19.mathdro.id/api'

const Map = () => {
  const [viewPort, setViewPort] = useState({
    width: '100vw',
    height: '60vh',
    latitude: 48.856613,
    longitude: 2.352222,
    zoom: 1.85,
  });

  return (
    <ReactMapGl
      {...viewPort}
      mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_TOKEN}
      mapStyle="mapbox://styles/mbrad26/ckdr7xcji0pq619oo5yj3ovuo"
      onViewportChange={newViewport => setViewPort(newViewport)}
    >
      <Marker
        latitude={48}
        longitude={2}
      >
        <button>Marker</button>
      </Marker>
    </ReactMapGl>
  );
};

const App = () => {
  const [data, setData] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [isError, setIsError] = useState(false)

  const fetchData = async () => {
    setIsLoading(true);

    try {
      const result = await axios.get(url);
      // console.log(result.data.confirmed.value);
      setData(result.data);
      setIsLoading(false)
    } catch (e) {
      console.log(e);
      setIsError(true);
    }
  };

  console.log("DATA: ", data);

  useEffect(() => {
    fetchData();
  }, [url]);

  return (
    <div className="App">
      <h2>Covid-19 Tracker</h2>
      <Map />
      {isLoading
        ? <h3>Loading ...</h3>
        : data.confirmed ? (
          <>
            <p>Confirmed:<span> {data.confirmed.value.toLocaleString()}</span></p>
            <p>Deaths:<span> {data.deaths.value.toLocaleString()}</span></p>
            <p>Recovered:<span> {data.recovered.value.toLocaleString()}</span></p>
          </>
        )
        : null
      }
      {isError && <h3>Something went wrong ...</h3>}
    </div>
  );
};

export default App;
