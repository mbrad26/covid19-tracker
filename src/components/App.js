import React from 'react';
import Map from './Map';
import DataCards from './DataCards';
import './App.css';

const App = () => {
  return (
    <div className="App">
      <h2>Covid-19 Tracker</h2>
      <Map />
      <DataCards />
    </div>
  );
};

export default App;
