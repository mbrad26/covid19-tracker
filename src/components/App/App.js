import React from 'react';
import Map from '../Map/Map';
import DataCards from '../DataCards/DataCards.js';
import './App.css';

const App = () => {
  return (
    <div className="App">
      <h2>Covid-19 Tracker</h2>
      <Map />
      <DataCards />
    </div>
  );
};

export default App;
