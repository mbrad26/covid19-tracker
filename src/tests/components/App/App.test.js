import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import store from '../../../store/index.js';
import App from '../../../components/App/App.js';

describe('App', () => {
  it('renders snapshot', () => {
    const { container } = render(
      <Provider store={store} >
        <App />
      </Provider>
      );

    expect(container.firstChild).toMatchSnapshot();
  });
});
