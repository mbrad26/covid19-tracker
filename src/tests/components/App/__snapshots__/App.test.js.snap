// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders snapshot 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-lg-12 col-md-12"
      id="map-container"
    >
      <div
        class=""
        id="map-component"
      >
        <div
          id="table-container"
        >
          <div
            class="spinner-border text-dark"
            id="spinner"
            role="status"
          />
        </div>
        <div
          style="position: relative; width: 593.92px; height: 499.20000000000005px; cursor: grab; outline: none;"
          tabindex="0"
        >
          <div
            style="position: relative; width: 100%; height: 100%;"
          >
            <div
              class=""
              style="position: absolute; width: 100%; height: 100%; overflow: hidden; visibility: inherit;"
            />
            <div
              style="overflow: visible; height: 0px; width: 0px;"
            />
            <div
              class="resize-triggers"
            >
              <div
                class="expand-trigger"
              >
                <div
                  style="width: 1px; height: 1px;"
                />
              </div>
              <div
                class="contract-trigger"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
    id="data-nhs-news"
  >
    <div
      class="col-xl-9 col-lg-12 col-md-7"
      id="data-cards-container"
    >
      <div>
        <div
          class="spinner-border text-dark"
          id="spinner-cards"
          role="status"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-12"
          id="nhs-container"
        >
          <div
            id="nhs"
          >
            <div
              class="spinner-border text-dark"
              id="spinner-nhs"
              role="status"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-xl-3 col-lg-12 col-md-5"
      id="news-container"
    >
      <div
        class="news row"
      >
        <div
          class="spinner-border text-dark"
          id="spinner-news"
          role="status"
        />
      </div>
    </div>
  </div>
</div>
`;
